compiler/semcall.nim

```diff
@@ -577,17 +578,32 @@ proc semResolvedCall(c: PContext, x: TCandidate,
       result.typ = newTypeS(x.fauxMatch, c)
       if result.typ.kind == tyError: incl result.typ.flags, tfCheckedForDestructor
     return
+
+  template buildBindings(x: TCandidate, expectedType: PType): TIdTable =
```

Comment:
Turn this into a helper proc please.

Reply:
Sure, should it stay in the same location or be moved somewhere else as well?

Reply:
Move it out so that we don't get a `.closure` accidentically (they are kinda slow and we avoid them in the compiler if we can).